exasol_sagemaker_extension/lua/src/autopilot_endpoint_deployment.lua

```diff
@@ -70,6 +71,13 @@ function main(
 		instance_type, instance_count, aws_s3_connection, aws_region)
 	local script_schema_name = exa.meta.script_schema
 
+	if not validate_input.is_autopilot_endpoint_name_valid(endpoint_name) then
+		local error_obj = exaerror.create("E-SME-11",
+				"Invalid endpoint name " ..job_name
+		):add_mitigations("The name of endpoint should match the following pattern: ^[a-zA-Z0-9]")
```

Comment:
I think the correct one is,

` ^[a-zA-Z0-9]([a-zA-Z0-9]){0,62}`

However, the aws [documentation](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateEndpoint.html) specifies

` ^[a-zA-Z0-9](-*[a-zA-Z0-9]){0,62}`

so with a dash "-" allowed. However, dashes are not a great thing in the DB, Maybe, we can think about allowing underscore and translate it into a dash, later. Similar is true for SageMaker Job Names (`^[a-zA-Z0-9](-*[a-zA-Z0-9]){0,31}`), see https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateAutoMLJob.html#API_CreateAutoMLJob_RequestSyntax

Reply:
yes, that's the main reason I opened this ticket anyway:
aws does not accept underscore, dash is not alowed in sql side. 

I thought it might be better to add some restrictions to naming rather than translating not-allowed chars

Reply:
with delimited (quoted) identifiers, dash is actually allowed in SQL https://docs.exasol.com/sql_references/basiclanguageelements.htm, but we can keep the restriction